component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -911,7 +913,7 @@ protected void processEmailOTPFlow(HttpServletRequest request, HttpServletRespon
             if (!context.isRetrying()
                     || (context.isRetrying()
                     && Boolean.parseBoolean(request.getParameter(EmailOTPAuthenticatorConstants.RESEND)))
-                    || (context.isRetrying() &&
+                    || (!isOTPResendingDisabledOnFailure(context) && context.isRetrying() &&
```

Comment:
Shall we take below part to another method too? Just a readability improvement. Something like isOTPExpired()